src/ctap/data_formats.rs

```diff
@@ -1166,6 +1166,13 @@ mod test {
         let policy_error = CredentialProtectionPolicy::try_from(cbor_policy_error);
         let expected_error = Err(Ctap2StatusCode::CTAP2_ERR_CBOR_UNEXPECTED_TYPE);
         assert_eq!(policy_error, expected_error);
+
+        for policy_literal in 1..=3 {
```

Comment:
We still have room for errors by hardcoding the last index. One solution would be to add a last enum value (always at the end without explicit value):

```rust
pub enum CredentialProtectionPolicy {
    UserVerificationOptional = 0x01,
    UserVerificationOptionalWithCredentialIdList = 0x02,
    UserVerificationRequired = 0x03,
    DoNotUseAndKeepLast,  // we can find a better name
}
```

Then we can loop with `for policy_literal in 1..CredentialProtectionPolicy::DoNotUseAndKeepLast as i64`.

Reply:
It looks like the `enum_iterator` crate achieves exactly what we need here: https://docs.rs/enum-iterator/0.6.0/enum_iterator/trait.IntoEnumIterator.html

Not sure whether we want to pull in another dependency, but the code boils down to the following macro: https://docs.rs/crate/enum-iterator-derive/0.6.0/source/src/lib.rs

Reply:
Right, the fact that we only need it in test makes it less problematic to add a dependency. This would also fix the issue with the transport enum.

Reply:
This seems like a complicated workaround for testing an enum. `DoNotUseAndKeepLast` should at least be behind a `#[cfg(test)]` ? We'd have to add this dummy entry into all enums that use integer indices. Because the risk is someone implementing an incorrect extension, and also not implementing a correct unit test for it?

Reply:
Yes you would need to add `#[cfg(test)]` and have it for all enums. There should not be more risks with this compared to hard-coding the last index.

However, I didn't realize that we only need the crate for testing, so I think this would be the best way to go. Because it also fixes the problem with the enums mapping to strings.

Reply:
That should work, yes. I added the crate for `enum-iterator` and use it conditioned on `#[cfg(test)]`.